src/parser/jobs/drk/ResourceSimulator.js

```diff
@@ -0,0 +1,599 @@
+				icon: ACTIONS.DARK_ARTS.icon,
+				content: <Fragment>
+					You wasted mana, and could have gotten more uses
+					of <ActionLink {...ACTIONS.DARK_ARTS}/> or other spenders during the fight
+					(minimum {DARK_ARTS_MANA_POTENCY} for {DARK_ARTS_MANA_COST} mana gained average combo potency increase each.)
+				</Fragment>,
+				severity: SEVERITY.MEDIUM,
+				why: <Fragment>
+					You wasted a total of {this._totalDroppedMana} from deaths and end of fight leftovers.
+				</Fragment>,
+			}))
+		}
+	}
+
+	static _aggregateSignedEntries(arr) {
+		const positiveAggregator = {}
+		const positiveEventAggregator = {}
+		const negativeAggregator = {}
+		const negativeEventAggregator = {}
+		const positiveOut = []
+		const negativeOut = []
+		let envNet = 0
+		while (arr.length > 0) {
+			const entry = arr.pop()
+			let name = 'Unknown'
+			if (entry.ability !== undefined) {
+				name = entry.ability.name
+			} else {
+				// non ability gains need to be aggregated and applied at the end
+				envNet += entry.value
+				continue
+			}
+			if (entry.value > 0) {
+				if (positiveAggregator.hasOwnProperty(name)) {
+					positiveAggregator[name] += entry.value
+				} else {
+					positiveAggregator[name] = entry.value
+					positiveEventAggregator[name] = entry.ability
+				}
+			} else if (negativeAggregator.hasOwnProperty(name)) {
+				negativeAggregator[name] += entry.value
+			} else {
+				negativeAggregator[name] = entry.value
+				negativeEventAggregator[name] = entry.ability
+			}
+		}
+		if (envNet !== 0) {
+			//pushed the combined environmental set into the appropriate box
+			if (envNet > 0) {
+				positiveOut.push({name: 'Initial Mana and Regen', ability: undefined, value: envNet})
+			} else {
+				negativeOut.push({name: 'Initial Mana and Regen', ability: undefined, value: envNet})
+			}
+		}
+		for (const property in positiveAggregator) {
+			positiveOut.push({name: property, ability: positiveEventAggregator[property], value: positiveAggregator[property]})
+		}
+		for (const property in negativeAggregator) {
+			negativeOut.push({name: property, ability: negativeEventAggregator[property], value: (negativeAggregator[property] * -1)})
+		}
+		return {positiveOut: positiveOut, negativeOut: negativeOut}
+	}
+
+	_filterResourceTrackingLists() {
+		const bloodDump = Resources._aggregateSignedEntries(this._blood_total_modifiers)
+		const manaDump = Resources._aggregateSignedEntries(this._mana_total_modifiers)
+		return {manaGenerators: manaDump.positiveOut, manaSpenders: manaDump.negativeOut, bloodGenerators: bloodDump.positiveOut, bloodSpenders: bloodDump.negativeOut}
+	}
+
+	_keyScumCounter = 0
+
+	_convertKVListToTable(list) {
+		const rows = []
+		while (list.length > 0) {
+			const entry = list.pop()
+			this._keyScumCounter += 1
+			const key = entry.name + '-' + this._keyScumCounter.toString()
+			//there's 100% a better fix for TBN to detect it as a status not an action, but frankly i'm too tired to care right now
```

Comment:
This sounds suspiciously like a TODO... Did you already do it and forget to remove the comment, or are you still too tired to care? :blobmorning:

Reply:
the line fixes some issue i can't remember, but should be improved at a later date. currently functional.  I think it had something to do with blackest night being a status.